lib/shopify_app/app_proxy_verification.rb

```diff
@@ -0,0 +1,32 @@
+module ShopifyApp
+  module AppProxyVerification
+    extend ActiveSupport::Concern
+
+    # Usage:
+    #   include ShopifyApp::AppProxyVerification
+    #   before_action :verify_proxy_request
+    def verify_proxy_request
+      return head :unauthorized unless query_string_valid?(request.query_string)
+    end
```

Comment:
Yes, that was the problem. The tests I wrote for this concern were simple - you just `include AppProxyVerification` directly in the test. But with an `included do` block that calls `before_action` and `skip_before_action`, the concern can only be included from a controller. But now that I've learned more about testing controllers, it will be simple to do. I'll add it in.